src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -77,7 +77,8 @@ public GoogleBucketResource getOrCreateBucketForFile(String possibleBucketName,
         BillingProfileModel billingProfile,
         String flightId) throws InterruptedException {
         final String datasetName = dataset.getName();
-        final GoogleRegion region = dataset.getDatasetSummary().getStorageResourceRegion(GoogleCloudResource.FIRESTORE);
+        final GoogleRegion region =
+            (GoogleRegion) dataset.getDatasetSummary().getStorageResourceRegion(GoogleCloudResource.FIRESTORE);
```

Comment:
Is this cast necessary? It looks like you could get rid of it, if you change all the code under this to use `CloudRegion` instead of `GoogleRegion`.

Reply:
I do need the cast since the underlying methods can return both

Reply:
But in this case every method called downstream can be converted to accept `CloudRegion`. So the code can be changed here to use `CloudRegion` instead of depend on `GoogleRegion`. This is because none of the code requires a feature that's not part of the i`CloudRegion` interface.